lib/daru/dataframe.rb

```diff
@@ -560,6 +560,11 @@ def replace_values old_values, new_value
       self
     end
 
+    def reset_index!
+      @index = Daru::Index.new(Array.new(size) { |i| i })
+      self
+    end
```

Comment:
Seems like I have implemented a very basic form of pandas' `reset_index` - of just resetting the `index`. I went through the complete doc and found that there are lots of functionalities of `reset_index`like taking care of `MultiIndex`, `Row`, `Column` and even `Series`. I'd prefer to isolate this function `reset_index` from this particular PR and continue in the later contributions as a separate PR.

Thanks for pointing this out! :smile: